src/OpenApi/test/Services/OpenApiSchemaService/OpenApiComponentService.RequestBodySchemas.cs

```diff
@@ -54,6 +55,56 @@ public async Task GetOpenApiRequestBody_GeneratesSchemaForPoco()
         });
     }
 
+    [Fact]
+    public async Task GetOpenApiRequestBody_RespectsRequiredAttributeOnBodyParameter()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapPost("/required-poco", ([Required] Todo todo) => { });
+        builder.MapPost("/non-required-poco", (Todo todo) => { });
+        builder.MapPost("/required-form", ([Required][FromForm] Todo todo) => { });
+        builder.MapPost("/non-required-form", ([FromForm] Todo todo) => { });
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            Assert.True(GetRequestBodyForPath(document, "/required-poco").Required);
+            Assert.False(GetRequestBodyForPath(document, "/non-required-poco").Required);
+            Assert.True(GetRequestBodyForPath(document, "/required-form").Required);
+            Assert.False(GetRequestBodyForPath(document, "/non-required-form").Required);
+        });
+
+        static OpenApiRequestBody GetRequestBodyForPath(OpenApiDocument document, string path)
+        {
+            var operation = document.Paths[path].Operations[OperationType.Post];
+            return operation.RequestBody;
+        }
+    }
+
+    [Fact]
+    public async Task GetOpenApiRequestBody_RespectsRequiredAttributeOnBodyProperties()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapPost("/required-properties", (RequiredTodo todo) => { });
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            var operation = document.Paths["/required-properties"].Operations[OperationType.Post];
+            var requestBody = operation.RequestBody;
+            var content = Assert.Single(requestBody.Content);
+            var schema = content.Value.Schema;
+            Assert.Collection(schema.Required,
```

Comment:
nit: `Assignee` should be checked for not required.

Reply:
The fact that it is not in the `schema.Required` list is indication enough but I can add a specific check to make it clearer.